texlive: provide repstopdf

veprbl opened the pull request:
~~It is just a symlink to epstopdf. epstopdf will enable restricted mode
if exectuted as repstopdf.~~ Usually repstopdf is just a symlink to epstopdf. This will not work with our tricky wrappers. A care must be taken to ensure that restricted mode gets enabled.

###### Motivation for this change

Fix #49405

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

veprbl commented:
/cc @jabranham @7c6f434c

veprbl commented:
@GrahamcOfBorg build texlive.combined.scheme-minimal texlive.combined.scheme-full

GrahamcOfBorg commented:
<!--REQUEST_ID=cc8c9137-6ec8-41eb-af9a-f27df1b65032-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=999dd465-b2b7-40e7-be27-95c9e476c83e)

Attempted: texlive.combined.scheme-minimal

The following builds were skipped because they don't evaluate on aarch64-linux: texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
'/nix/store/yyx6x9h8xdhzxxnr114s14sxph2l7h29-texlive-combined-minimal-2018/bin/fmtutil-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-sys.sh'
'/nix/store/yyx6x9h8xdhzxxnr114s14sxph2l7h29-texlive-combined-minimal-2018/bin/fmtutil-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-user.sh'
'/nix/store/yyx6x9h8xdhzxxnr114s14sxph2l7h29-texlive-combined-minimal-2018/bin/rungs' -> '/nix/store/qhmxgxmcajhdvwiy1l1nb46wg22sijmz-texlive-texlive-scripts-2018/scripts/texlive/rungs.tlu'
'/nix/store/yyx6x9h8xdhzxxnr114s14sxph2l7h29-texlive-combined-minimal-2018/bin/updmap-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-sys.sh'
'/nix/store/yyx6x9h8xdhzxxnr114s14sxph2l7h29-texlive-combined-minimal-2018/bin/updmap-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-user.sh'
Wrapping './bin/fmtutil-user' and patching shebang '/bin/sh'
Wrapping './bin/rungs' and patching shebang '/usr/bin/env texlua'
Wrapping './bin/simpdftex' and patching shebang '/bin/sh'
Wrapping './bin/updmap-user' and patching shebang '/bin/sh'
/nix/store/yyx6x9h8xdhzxxnr114s14sxph2l7h29-texlive-combined-minimal-2018
```
</p></details>

7c6f434c commented:
@vcunat so was https://github.com/NixOS/nixpkgs/commit/7aa140488aa3e3c3557cafa1bda0ad8f3bad5aef ever applied to `master`?

GrahamcOfBorg commented:
<!--REQUEST_ID=cc8c9137-6ec8-41eb-af9a-f27df1b65032-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=c848a25f-5c87-43e6-8a81-617ca2b7c2d1)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf' saved in '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/70039cb0a869385816e10015d9d8adad.lua'
system          | lua | compiling '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/70039cb0a869385816e10015d9d8adad.lua' into '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/70039cb0a869385816e10015d9d8adad.luc'
system          | lua | dumping '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/70039cb0a869385816e10015d9d8adad.lua' into '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/70039cb0a869385816e10015d9d8adad.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/70039cb0a869385816e10015d9d8adad.luc'
mtxrun          |
mtxrun          | elapsed lua time: 2.993 seconds

/nix/store/4s5m6mcc48ln4nnfqv8zz18bl8pra85l-texlive-combined-minimal-2018
/nix/store/zq4573dqrgw0n89j8c4wwwrggxfn0fx4-texlive-combined-full-2018
```
</p></details>

veprbl commented:
Judging by 7aa140488aa3e3c3557cafa1bda0ad8f3bad5aef this must be broken

veprbl commented:
Actually, the 7aa140488aa3e3c3557cafa1bda0ad8f3bad5aef doesn't work as well.

GrahamcOfBorg commented:
<!--REQUEST_ID=cc8c9137-6ec8-41eb-af9a-f27df1b65032-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=fc4fe523-89c6-473e-a571-dccb00f2d08d)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf' saved in '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d0d09337caf5954c68c1aa590de77f11.lua'
system          | lua | compiling '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d0d09337caf5954c68c1aa590de77f11.lua' into '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d0d09337caf5954c68c1aa590de77f11.luc'
system          | lua | dumping '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d0d09337caf5954c68c1aa590de77f11.lua' into '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d0d09337caf5954c68c1aa590de77f11.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d0d09337caf5954c68c1aa590de77f11.luc'
mtxrun          |
mtxrun          | elapsed lua time: 4.847 seconds

/nix/store/ziaspb0dywba64gdxlbsjzxgz2m2fdrr-texlive-combined-minimal-2018
/nix/store/549q90zklf8xbbfjk60wj8rryi73jnky-texlive-combined-full-2018
```
</p></details>

veprbl committed:
texlive: provide repstopdf

repstopdf is supposed to be a symlink to epstopdf. Then epstopdf looks
at "$0" to detect when restricted mode needs to be enabled.  Unfortunately
our wrapper will drop all intermediate symlinks, which messes up "$0".
Restricted mode appears to be a security feature, so a test is
introduced to verify that the wrapper works as expected.

veprbl commented:
Please take a look at updated version.

veprbl commented:
@GrahamcOfBorg build texlive.combined.scheme-minimal texlive.combined.scheme-full

GrahamcOfBorg commented:
<!--REQUEST_ID=aa77da64-bc29-41df-9726-2cb570caf017-->
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=d14857c1-3a3d-4bdf-b595-a758f4f385a7)

Attempted: texlive.combined.scheme-minimal

The following builds were skipped because they don't evaluate on aarch64-linux: texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
  /nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/share/texmf-config/web2c/updmap.cfg
'/nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/bin/simpdftex' -> '/nix/store/xvyrbd7y1x77lp19rsdwm4hw1krdivpa-texlive-pdftex-2018/scripts/simpdftex/simpdftex'
'/nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/bin/fmtutil-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-sys.sh'
'/nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/bin/fmtutil-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-user.sh'
'/nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/bin/rungs' -> '/nix/store/qhmxgxmcajhdvwiy1l1nb46wg22sijmz-texlive-texlive-scripts-2018/scripts/texlive/rungs.tlu'
'/nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/bin/updmap-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-sys.sh'
'/nix/store/nrr46hgirrn6b0mps7qnjp665f3vib15-texlive-combined-minimal-2018/bin/updmap-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-user.sh'
Testing restricted mode
builder for '/nix/store/99jk3w6n25rdbcwxs63r6ls191ja9mhf-texlive-combined-minimal-2018.drv' failed with exit code 1
error: build of '/nix/store/99jk3w6n25rdbcwxs63r6ls191ja9mhf-texlive-combined-minimal-2018.drv' failed
```
</p></details>

veprbl commented:
@GrahamcOfBorg build texlive.combined.scheme-minimal texlive.combined.scheme-full

GrahamcOfBorg commented:
<!--REQUEST_ID=15bbde29-d1e8-4f1d-b7f4-d6012456131b-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=c9107d5c-dfc2-48b2-afb7-4c25c2227b4f)

Attempted: texlive.combined.scheme-minimal

The following builds were skipped because they don't evaluate on aarch64-linux: texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
'/nix/store/f2kywpz30bpfz93ikqgi6svmpvqady8d-texlive-combined-minimal-2018/bin/fmtutil-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-sys.sh'
'/nix/store/f2kywpz30bpfz93ikqgi6svmpvqady8d-texlive-combined-minimal-2018/bin/fmtutil-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-user.sh'
'/nix/store/f2kywpz30bpfz93ikqgi6svmpvqady8d-texlive-combined-minimal-2018/bin/rungs' -> '/nix/store/qhmxgxmcajhdvwiy1l1nb46wg22sijmz-texlive-texlive-scripts-2018/scripts/texlive/rungs.tlu'
'/nix/store/f2kywpz30bpfz93ikqgi6svmpvqady8d-texlive-combined-minimal-2018/bin/updmap-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-sys.sh'
'/nix/store/f2kywpz30bpfz93ikqgi6svmpvqady8d-texlive-combined-minimal-2018/bin/updmap-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-user.sh'
Wrapping './bin/fmtutil-user' and patching shebang '/bin/sh'
Wrapping './bin/rungs' and patching shebang '/usr/bin/env texlua'
Wrapping './bin/simpdftex' and patching shebang '/bin/sh'
Wrapping './bin/updmap-user' and patching shebang '/bin/sh'
/nix/store/f2kywpz30bpfz93ikqgi6svmpvqady8d-texlive-combined-minimal-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=15bbde29-d1e8-4f1d-b7f4-d6012456131b-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=b53bb8a8-5f6f-4e61-b341-db665f11f3f7)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf' saved in '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/487e6d26203893cf889c2e1a4f3d7d1d.lua'
system          | lua | compiling '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/487e6d26203893cf889c2e1a4f3d7d1d.lua' into '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/487e6d26203893cf889c2e1a4f3d7d1d.luc'
system          | lua | dumping '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/487e6d26203893cf889c2e1a4f3d7d1d.lua' into '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/487e6d26203893cf889c2e1a4f3d7d1d.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/487e6d26203893cf889c2e1a4f3d7d1d.luc'
mtxrun          |
mtxrun          | elapsed lua time: 2.785 seconds

/nix/store/9g295s8lljjd8v5a0gcrg8sa9p0pbhw1-texlive-combined-minimal-2018
/nix/store/6vhp91sk5l8ww6i2zkj7aj84mfn4npww-texlive-combined-full-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=15bbde29-d1e8-4f1d-b7f4-d6012456131b-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=f245a9bb-82e3-43c1-8c8f-c17ae6f06ebf)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/htmex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/htmex.sh'
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/httex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/httex.sh'
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/httexi' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/httexi.sh'
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/htxelatex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/htxelatex.sh'
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/htxetex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/htxetex.sh'
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/mk4ht' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/mk4ht.pl'
'/nix/store/k2jivifb7pm8jx4yyf9lwppc3pgfc2ji-texlive-combined-full-2018/bin/xhlatex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/xhlatex.sh'
Testing restricted mode for {,r}epstopdf
builder for '/nix/store/0k2l8np8m5fk7dn8lyfcccfg6gw5c312-texlive-combined-full-2018.drv' failed with exit code 1
error: build of '/nix/store/0k2l8np8m5fk7dn8lyfcccfg6gw5c312-texlive-combined-full-2018.drv' failed
```
</p></details>

veprbl commented:
@GrahamcOfBorg build texlive.combined.scheme-minimal texlive.combined.scheme-full

GrahamcOfBorg commented:
<!--REQUEST_ID=852c9c33-aae1-43d5-85db-2f0f71fd2453-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=e11d36c1-d3b1-4893-b344-daeed51295cf)

Attempted: texlive.combined.scheme-minimal

The following builds were skipped because they don't evaluate on aarch64-linux: texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
'/nix/store/dk4d0898pzk84ds67z811sh3frcqqn5a-texlive-combined-minimal-2018/bin/fmtutil-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-sys.sh'
'/nix/store/dk4d0898pzk84ds67z811sh3frcqqn5a-texlive-combined-minimal-2018/bin/fmtutil-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-user.sh'
'/nix/store/dk4d0898pzk84ds67z811sh3frcqqn5a-texlive-combined-minimal-2018/bin/rungs' -> '/nix/store/qhmxgxmcajhdvwiy1l1nb46wg22sijmz-texlive-texlive-scripts-2018/scripts/texlive/rungs.tlu'
'/nix/store/dk4d0898pzk84ds67z811sh3frcqqn5a-texlive-combined-minimal-2018/bin/updmap-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-sys.sh'
'/nix/store/dk4d0898pzk84ds67z811sh3frcqqn5a-texlive-combined-minimal-2018/bin/updmap-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-user.sh'
Wrapping './bin/fmtutil-user' and patching shebang '/bin/sh'
Wrapping './bin/rungs' and patching shebang '/usr/bin/env texlua'
Wrapping './bin/simpdftex' and patching shebang '/bin/sh'
Wrapping './bin/updmap-user' and patching shebang '/bin/sh'
/nix/store/dk4d0898pzk84ds67z811sh3frcqqn5a-texlive-combined-minimal-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=852c9c33-aae1-43d5-85db-2f0f71fd2453-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=7f03c7eb-595c-4c4e-a8f7-8c2c3f9a0379)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf' saved in '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d6c5acb2b232f09407fec96022fa748b.lua'
system          | lua | compiling '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d6c5acb2b232f09407fec96022fa748b.lua' into '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d6c5acb2b232f09407fec96022fa748b.luc'
system          | lua | dumping '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d6c5acb2b232f09407fec96022fa748b.lua' into '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d6c5acb2b232f09407fec96022fa748b.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/d6c5acb2b232f09407fec96022fa748b.luc'
mtxrun          |
mtxrun          | elapsed lua time: 5.263 seconds

/nix/store/xr2yfr529pjcys96j3xacqlsf3acs5gx-texlive-combined-minimal-2018
/nix/store/pfag3lwgjqpms5rc126sbpab5r6mi1ay-texlive-combined-full-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=852c9c33-aae1-43d5-85db-2f0f71fd2453-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=8b7edcf9-dfb7-4fec-aa65-93cf85249b51)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
'/nix/store/pxpqkj5ww1zw04kcpk525dbpr46cx0yg-texlive-combined-full-2018/bin/htxelatex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/htxelatex.sh'
'/nix/store/pxpqkj5ww1zw04kcpk525dbpr46cx0yg-texlive-combined-full-2018/bin/htxetex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/htxetex.sh'
'/nix/store/pxpqkj5ww1zw04kcpk525dbpr46cx0yg-texlive-combined-full-2018/bin/mk4ht' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/mk4ht.pl'
'/nix/store/pxpqkj5ww1zw04kcpk525dbpr46cx0yg-texlive-combined-full-2018/bin/xhlatex' -> '/nix/store/nbg2myaf5z7qzmvqiiw3zf2wxbml533w-texlive-tex4ht-2018/scripts/tex4ht/xhlatex.sh'
Testing restricted mode for {,r}epstopdf
/nix/store/3h4j4qk4996cch5xsxldng4r6a5i4cwb-stdenv-linux/setup: /nix/store/pxpqkj5ww1zw04kcpk525dbpr46cx0yg-texlive-combined-full-2018/bin/epstopdf: /usr/bin/env: bad interpreter: No such file or directory
foo
/nix/store/3h4j4qk4996cch5xsxldng4r6a5i4cwb-stdenv-linux/setup: /nix/store/pxpqkj5ww1zw04kcpk525dbpr46cx0yg-texlive-combined-full-2018/bin/repstopdf: /usr/bin/env: bad interpreter: No such file or directory
builder for '/nix/store/xp319mg8rwjssrxygna21cp9dn6xljln-texlive-combined-full-2018.drv' failed with exit code 1
error: build of '/nix/store/xp319mg8rwjssrxygna21cp9dn6xljln-texlive-combined-full-2018.drv' failed
```
</p></details>

veprbl commented:
@GrahamcOfBorg build texlive.combined.scheme-minimal texlive.combined.scheme-full

GrahamcOfBorg commented:
<!--REQUEST_ID=71779a94-e121-4e0d-967f-ad56376012ff-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=57c5fa47-db2e-43eb-bf09-b1f3fd475e8e)

Attempted: texlive.combined.scheme-minimal

The following builds were skipped because they don't evaluate on aarch64-linux: texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
'/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018/bin/fmtutil-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-sys.sh'
'/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018/bin/fmtutil-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/fmtutil-user.sh'
'/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018/bin/rungs' -> '/nix/store/qhmxgxmcajhdvwiy1l1nb46wg22sijmz-texlive-texlive-scripts-2018/scripts/texlive/rungs.tlu'
'/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018/bin/updmap-sys' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-sys.sh'
'/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018/bin/updmap-user' -> '/nix/store/inkiad0nm0764a9dvk22yshdcj3rcn11-texlive-tetex-3.0/scripts/texlive/updmap-user.sh'
Wrapping './bin/fmtutil-user' and patching shebang '/bin/sh'
Wrapping './bin/rungs' and patching shebang '/usr/bin/env texlua'
Wrapping './bin/simpdftex' and patching shebang '/bin/sh'
Wrapping './bin/updmap-user' and patching shebang '/bin/sh'
/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=71779a94-e121-4e0d-967f-ad56376012ff-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=53f11ba9-484d-4ffb-be54-49d4b098bcd7)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf' saved in '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/9c756dd6623eb3f13654b00758862d89.lua'
system          | lua | compiling '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/9c756dd6623eb3f13654b00758862d89.lua' into '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/9c756dd6623eb3f13654b00758862d89.luc'
system          | lua | dumping '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/9c756dd6623eb3f13654b00758862d89.lua' into '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/9c756dd6623eb3f13654b00758862d89.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/9c756dd6623eb3f13654b00758862d89.luc'
mtxrun          |
mtxrun          | elapsed lua time: 5.267 seconds

/nix/store/90vg6k4c6ddxllhivwq6pdmh9142b41w-texlive-combined-minimal-2018
/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=71779a94-e121-4e0d-967f-ad56376012ff-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=53608b06-b9a2-4dda-a530-527b989ed9ad)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf' saved in '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.lua'
system          | lua | compiling '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.lua' into '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.luc'
system          | lua | dumping '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.lua' into '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.luc'
mtxrun          |
mtxrun          | elapsed lua time: 4.822 seconds

/nix/store/rs296pad7pprdglc7mvnx00mmwx7i64k-texlive-combined-minimal-2018
/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=aa77da64-bc29-41df-9726-2cb570caf017-->
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=633bdadc-3e2b-4f23-b54c-d69f13c34a20)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caches | hashing tree '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf', hash '84e148dd58cd4a85174d7197ed51e48a'
resolvers       | caching | preparing 'files' for '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf' saved in '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/84e148dd58cd4a85174d7197ed51e48a.lua'
system          | lua | compiling '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/84e148dd58cd4a85174d7197ed51e48a.lua' into '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/84e148dd58cd4a85174d7197ed51e48a.luc'
system          | lua | dumping '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/84e148dd58cd4a85174d7197ed51e48a.lua' into '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/84e148dd58cd4a85174d7197ed51e48a.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/c0zqswdp2pfhdhs5n2nk5ilpama8fg2h-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/84e148dd58cd4a85174d7197ed51e48a.luc'
mtxrun          |
mtxrun          | elapsed lua time: 5.245 seconds

error: build of '/nix/store/5x2l2l0p973zfy0ivl17ppw8fbxzaqzi-texlive-combined-minimal-2018.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=aa77da64-bc29-41df-9726-2cb570caf017-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=263e5b22-4105-4927-899a-f21a1c22837e)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
building '/nix/store/h93f1x2yirjji0d1ln0b9g47ahdw9nws-zxjatype.tar.xz.drv'...

trying https://cat3.de/texlive-2018/tlnet/archive/zxjatype.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  8408  100  8408    0     0  76436      0 --:--:-- --:--:-- --:--:-- 76436
building '/nix/store/mai3cg50lfr2m39gdn9kyihj587xys9f-texlive-zxjafont-0.4.drv'...
building '/nix/store/z4x38i2djxxcqr9qqkzspmyksvmycayd-texlive-zxjatype-0.6c.drv'...
cannot build derivation '/nix/store/s3n91q9qcgk7q8v4468g5byyqm0fkl5y-texlive-combined-full-2018.drv': 1 dependencies couldn't be built
error: build of '/nix/store/bv9dav0h7kdrsl7s2vvxrv6cjfi08fya-texlive-combined-minimal-2018.drv', '/nix/store/s3n91q9qcgk7q8v4468g5byyqm0fkl5y-texlive-combined-full-2018.drv' failed
```
</p></details>

veprbl commented:
@GrahamcOfBorg build texlive.combined.scheme-minimal texlive.combined.scheme-full

GrahamcOfBorg commented:
<!--REQUEST_ID=e9a53b02-3973-410e-854c-2d99cac0b24c-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=a36abeb6-d4f2-48d5-b422-dff0b88039d3)

Attempted: texlive.combined.scheme-minimal

The following builds were skipped because they don't evaluate on aarch64-linux: texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.


/nix/store/g9nrwzhm1sajyjycv1v9d1cipfwix6gh-texlive-combined-minimal-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e9a53b02-3973-410e-854c-2d99cac0b24c-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=1dda73ea-95df-4e39-b35a-c762610f4b41)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/90vg6k4c6ddxllhivwq6pdmh9142b41w-texlive-combined-minimal-2018
/nix/store/cb7iq5iiqhyhpvpll8gwaz0qyg80yzg5-texlive-combined-full-2018
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=e9a53b02-3973-410e-854c-2d99cac0b24c-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.49416&attempt_id=a6277a3d-b828-4ea5-9649-9a958521ddf8)

Attempted: texlive.combined.scheme-minimal, texlive.combined.scheme-full

<details><summary>Partial log (click to expand)</summary><p>

```
resolvers       | caching | preparing 'files' for '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf'
resolvers       | caching | category 'files', cachename '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf' saved in '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.lua'
system          | lua | compiling '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.lua' into '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.luc'
system          | lua | dumping '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.lua' into '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.luc' stripped
resolvers       | caching | 'files' compiled to '/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018/share/texmf-var/luatex-cache/context/f919750633122078d706dd10f1cfd39b/trees/a412384cedfb094f9c6f0d682db20d62.luc'
mtxrun          |
mtxrun          | elapsed lua time: 4.835 seconds

/nix/store/rs296pad7pprdglc7mvnx00mmwx7i64k-texlive-combined-minimal-2018
/nix/store/khpgixdp2nvni270mkv97sskpv85ahps-texlive-combined-full-2018
```
</p></details>

vcunat commented:
@7c6f434c: no, it has never been applied.

7c6f434c commented:
@vcunat well, I did check it back in 2017 and posted the result in the old issue… Anyway, this PR now does take into account the code from the old patch, so do we declare the current state ready to merge?

7c6f434c commented:
OK, the current state improves the situation, and now I don't see anything obvious to cross-check now.
